pkg/controller/replicaset/replica_set.go

```diff
@@ -802,6 +802,10 @@ func getPodsToDelete(filteredPods, relatedPods []*v1.Pod, diff int) []*v1.Pod {
 	// diff will always be <= len(filteredPods), so not need to handle > case.
 	if diff < len(filteredPods) {
 		podsWithRanks := getPodsRankedByRelatedPodsOnSameNode(filteredPods, relatedPods)
+		// First obtain a pseudo-random ordering by sorting by pod UID
+		sort.Slice(podsWithRanks.Pods, func(i, j int) bool {
```

Comment:
Why not just add the comparison in the existing `Less`? I feel like that should perform better.

Reply:
Should that come before or after the timestamp comparison?

Reply:
it's a comparison criteria with the lowest priority, so after.

Reply:
in that case I should also update [`logarithmicOrAfterZero`](https://github.com/kubernetes/kubernetes/pull/99212/files#diff-8128d56d17ba6f10e337cf42856a93fcfaf5bce5b7efbc6d9a371bb5c04c7098R908) to check if the 2 ranks are equal (and continue), rather than just returning... or I can add the UID check right into `logarithmicOrAfterZero`. Wdyt?

Reply:
probably better to be explicit about UIDs by having that comparison outside of timestamp comparison.